Don't try to send notification SMS when Nexmo is not configured [0.5]

### Bug Description
Following error leads to the error "Server connection lost" in the UI. Users have to close the browser and relogin in order to use SORMAS
```
{
  "Timestamp": "2021-04-15T15:20:00.188+0200",
  "Level": "WARNUNG",
  "Version": "Payara 5.2020",
  "LoggerName": "javax.enterprise.ejb.container",
  "ThreadID": "250",
  "ThreadName": "__ejb-thread-pool2",
  "TimeMillis": "1618492800188",
  "LevelValue": "900",
  "Throwable": {
    "Exception": "Something went wrong while executing the HTTP request: api.nexmo.com: Name or service not known.",
    "StackTrace": "javax.ejb.EJBException: Something went wrong while executing the HTTP request: api.nexmo.com: Name or service not known.\n\tat com.sun.ejb.containers.EJBContainerTransactionManager.processSystemException(EJBContainerTransactionManager.java:723)\n\tat com.sun.ejb.containers.EJBContainerTransactionManager.completeNewTx(EJBContainerTransactionManager.java:652)\n\tat com.sun.ejb.containers.EJBContainerTransactionManager.postInvokeTx(EJBContainerTransactionManager.java:482)\n\tat com.sun.ejb.containers.BaseContainer.postInvokeTx(BaseContainer.java:4562)\n\tat com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2111)\n\tat com.sun.ejb.containers.EjbAsyncTask.call(EjbAsyncTask.java:114)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\nCaused by: com.nexmo.client.NexmoMethodFailedException: Something went wrong while executing the HTTP request: api.nexmo.com: Name or service not known.\n\tat com.nexmo.client.AbstractMethod.execute(AbstractMethod.java:117)\n\tat com.nexmo.client.insight.InsightClient.getStandardNumberInsight(InsightClient.java:151)\n\tat com.nexmo.client.insight.InsightClient.getStandardNumberInsight(InsightClient.java:102)\n\tat de.symeda.sormas.backend.common.messaging.SmsService.sendSms(SmsService.java:64)\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n\tat org.glassfish.ejb.security.application.EJBSecurityManager.runMethod(EJBSecurityManager.java:588)\n\tat org.glassfish.ejb.security.application.EJBSecurityManager.invoke(EJBSecurityManager.java:408)\n\tat com.sun.ejb.containers.BaseContainer.invokeBeanMethod(BaseContainer.java:4796)\n\tat com.sun.ejb.EjbInvocation.invokeBeanMethod(EjbInvocation.java:665)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)\n\tat com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)\n\tat de.symeda.sormas.backend.util.PerformanceLoggingInterceptor.logInvokeDuration(PerformanceLoggingInterceptor.java:37)\n\tat jdk.internal.reflect.GeneratedMethodAccessor222.invoke(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)\n\tat com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)\n\tat com.sun.ejb.containers.interceptors.SystemInterceptorProxy.doCall(SystemInterceptorProxy.java:163)\n\tat com.sun.ejb.containers.interceptors.SystemInterceptorProxy.aroundInvoke(SystemInterceptorProxy.java:140)\n\tat jdk.internal.reflect.GeneratedMethodAccessor221.invoke(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)\n\tat com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)\n\tat org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:81)\n\tat org.jboss.weld.module.ejb.SessionBeanInterceptor.aroundInvoke(SessionBeanInterceptor.java:52)\n\tat jdk.internal.reflect.GeneratedMethodAccessor199.invoke(Unknown Source)\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)\n\tat com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)\n\tat com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:375)\n\tat com.sun.ejb.containers.BaseContainer.__intercept(BaseContainer.java:4768)\n\tat com.sun.ejb.containers.BaseContainer.intercept(BaseContainer.java:4756)\n\tat com.sun.ejb.containers.EjbAsyncTask.call(EjbAsyncTask.java:101)\n\t... 4 more\nCaused by: java.net.UnknownHostException: api.nexmo.com: Name or service not known\n\tat java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)\n\tat java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)\n\tat java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1519)\n\tat java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)\n\tat java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)\n\tat java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)\n\tat java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)\n\tat org.apache.http.impl.conn.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:45)\n\tat org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:112)\n\tat org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)\n\tat org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)\n\tat org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)\n\tat org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)\n\tat org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)\n\tat org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)\n\tat org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)\n\tat org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)\n\tat org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)\n\tat org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)\n\tat com.nexmo.client.AbstractMethod.execute(AbstractMethod.java:103)\n\t... 43 more\n"
  }
}
```

### System Details
* SORMAS version: 1.57 & 1.58 (Dockerized)

### Additional Information
Server is behind a proxy and SORMAS is using a proxy configuration

--- a/sormas-backend/src/main/java/de/symeda/sormas/backend/common/messaging/MessagingService.java
+++ b/sormas-backend/src/main/java/de/symeda/sormas/backend/common/messaging/MessagingService.java
@@ -30,10 +30,10 @@ import org.slf4j.LoggerFactory;
 
 import com.nexmo.client.NexmoClientException;
 
-import de.symeda.sormas.api.feature.FeatureType;
 import de.symeda.sormas.api.i18n.I18nProperties;
 import de.symeda.sormas.api.messaging.MessageType;
 import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.backend.common.ConfigFacadeEjb;
 import de.symeda.sormas.backend.feature.FeatureConfigurationFacadeEjb;
 import de.symeda.sormas.backend.person.Person;
 import de.symeda.sormas.backend.user.User;
@@ -88,6 +88,8 @@ public class MessagingService {
 	private ManualMessageLogService manualMessageLogService;
 	@EJB
 	private FeatureConfigurationFacadeEjb.FeatureConfigurationFacadeEjbLocal featureConfigurationFacade;
+	@EJB
+	private ConfigFacadeEjb.ConfigFacadeEjbLocal configFacade;
 
 	/**
 	 * Sends the message specified by the messageContent via mail and/or SMS, according to the messageTypes, to the specified recipient's
@@ -104,29 +106,19 @@ public class MessagingService {
 		sendMessage(recipient, I18nProperties.getEnumCaption(subject), messageContent, messageTypes);
 	}
 
-
 	/**
 	 * Sends the message specified by the messageContent via mail and/or SMS, according to the messageTypes, to the specified recipient's
 	 * email address and/or phone number. Logs an error if the email address or phone number is not set.
 	 */
-	public void sendMessage(
-			User recipient,
-			MessageSubject subject,
-			Object[] subjectParameters,
-			String messageContent,
-			MessageType... messageTypes)
-			throws NotificationDeliveryFailedException {
+	public void sendMessage(User recipient, MessageSubject subject, Object[] subjectParameters, String messageContent, MessageType... messageTypes)
+		throws NotificationDeliveryFailedException {
 
 		// Don't send notifications if the feature is disabled for the current MessageSubject
 		if (!featureConfigurationFacade.isFeatureEnabled(subject.getRelatedFeatureType())) {
 			return;
 		}
 
-		sendMessage(
-				recipient,
-				String.format(I18nProperties.getEnumCaption(subject), subjectParameters),
-				messageContent,
-				messageTypes);
+		sendMessage(recipient, String.format(I18nProperties.getEnumCaption(subject), subjectParameters), messageContent, messageTypes);
 	}
 
 	private void sendMessage(User recipient, String subject, String messageContent, MessageType... messageTypes)
@@ -167,16 +159,18 @@ public class MessagingService {
 		final String recipientType,
 		MessageType... messageTypes)
 		throws NotificationDeliveryFailedException {
+
+		boolean isSmsServiceSetUp = configFacade.isSmsServiceSetUp();
 		for (MessageType messageType : messageTypes) {
 			if (messageType == MessageType.EMAIL && DataHelper.isNullOrEmpty(emailAddress)) {
 				logger.info(String.format("Tried to send an email to a " + recipientType + " without an email address (UUID: %s).", recipientUuid));
-			} else if (messageType == MessageType.SMS && DataHelper.isNullOrEmpty(phoneNumber)) {
+			} else if (isSmsServiceSetUp && messageType == MessageType.SMS && DataHelper.isNullOrEmpty(phoneNumber)) {
 				logger.info(String.format("Tried to send an SMS to a " + recipientType + " without a phone number (UUID: %s).", recipientUuid));
 			} else {
 				try {
 					if (messageType == MessageType.EMAIL) {
 						emailService.sendEmail(emailAddress, subject, messageContent);
-					} else if (messageType == MessageType.SMS) {
+					} else if (isSmsServiceSetUp && messageType == MessageType.SMS) {
 						smsService.sendSms(phoneNumber, messageContent);
 					}
 				} catch (MessagingException e) {

--- a/sormas-backend/src/main/java/de/symeda/sormas/backend/common/messaging/SmsService.java
+++ b/sormas-backend/src/main/java/de/symeda/sormas/backend/common/messaging/SmsService.java
@@ -52,6 +52,10 @@ public class SmsService {
 	@Asynchronous
 	public void sendSms(String phoneNumber, String content) throws IOException, NexmoClientException, InvalidPhoneNumberException {
 
+		if (!configFacade.isSmsServiceSetUp()) {
+			return;
+		}
+
 		// Remove the initial + that indicates the beginning of the country code to match the Nexmo specification of allowed number formats
 		if (phoneNumber.startsWith("+")) {
 			phoneNumber = phoneNumber.substring(1);

